examples/frameworks/adk_demo/src/nat_adk_demo/location.py

```
@@ -0,0 +1,123 @@
+        if self.admin1 and self.admin1.casefold() != self.name.casefold():
+            parts.append(self.admin1)
+        if self.country:
+            parts.append(self.country)
+        return ", ".join(parts)
+
+
+def fetch_json(url: str, params: dict[str, str | int | float], timeout_seconds: float) -> dict[str, Any]:
+    """Fetch a JSON object using only the Python standard library."""
+
+    request_url = f"{url}?{urlencode(params)}"
+    request = Request(request_url, headers={"User-Agent": "nat-adk-demo/1.0"})
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)